### PR TITLE
Make create_pull_request get current branch and a title as args

### PR DESCRIPTION
**Pull request by RazvanDire**
For now, the title of the pr is the same as the branches name

### CODE REVIEW

**Comment by RazvanDire** [comment]
* **#26** <a href="https://app.graphite.dev/github/pr/costinsin/gi/26?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25** <a href="https://app.graphite.dev/github/pr/costinsin/gi/25?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24** <a href="https://app.graphite.dev/github/pr/costinsin/gi/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23** <a href="https://app.graphite.dev/github/pr/costinsin/gi/23?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RazvanDire and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/commands/submit.rs [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 pub async fn submit() -> Result<()> {
-    let (provider, owner, repo) = git_client::get_git_client()?.get_repository_info()?;
-    let provider_obj = provider_factory(&provider)?;
+    let git_client = git_client::get_git_client()?;
+
+    let branch = git_client
+        .get_current_branch()
+        .ok_or_eyre("Failed to get the current branch")
+        .suggestion("Check whether you are checked out onto a branch")?;
+    let title = branch.clone();
```
Comment (costinsin) [inline src/commands/submit.rs:13]: Do you need this clone id if you only immutably borrow title?

**Review by costinsin — APPROVED** [review]

File: src/commands/submit.rs (hunk 1 above)
Comment (RazvanDire) [reply src/commands/submit.rs:13]: I think not, but this is just a placeholder; in the future, the user will have the option to change the title of the PR

**Comment by RazvanDire** [comment]
### Merge activity

* **May 4, 9:51 AM EDT**: @RazvanDire started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/costinsin/gi/23).

**Pull request merged**